src/Sylius/Bundle/CoreBundle/Sitemap/Provider/ProductUrlProvider.php

```diff
@@ -66,7 +65,7 @@ public function generate()
 
         foreach ($products as $product) {
             $productUrl = $this->sitemapUrlFactory->createNew();
-            $localization = $this->router->generate($product, [], true);
+            $localization = $this->router->generate('sylius_shop_product_show', ['slug' => $product->getSlug()], true);
```

Comment:
This couples core with ShopBundle, but I will open a PR, which fixes that later.